### PR TITLE
bump ruff version

### PR DESCRIPTION
**Pull request by danieleades**
i'm not convinced all of these changes improve readability, though i think most of them do.

let me know if there's any that should be manually suppressed

### CODE REVIEW

File: sphinx/ext/doctest.py [hunk 1]
```diff
@@ -509,9 +509,7 @@ def run_setup_cleanup(runner: Any, testcodes: list[TestCode], what: Any) -> bool
             old_f = runner.failures
             self.type = 'exec'  # the snippet may contain multiple statements
             runner.run(sim_doctest, out=self._warn_out, clear_globs=False)
-            if runner.failures > old_f:
-                return False
-            return True
+            return not runner.failures > old_f
```
Comment (picnixz) [inline sphinx/ext/doctest.py]:
```suggestion
            return runner.failures <= old_f
```

File: sphinx/transforms/post_transforms/__init__.py [hunk 2]
```diff
@@ -47,10 +47,7 @@ def is_supported(self) -> bool:
         """Check this transform working for current builder."""
         if self.builders and self.app.builder.name not in self.builders:
             return False
-        if self.formats and self.app.builder.format not in self.formats:
-            return False
-
-        return True
+        return not (self.formats and self.app.builder.format not in self.formats)
```
Comment (picnixz) [inline sphinx/transforms/post_transforms/__init__.py]:
```suggestion
        return not self.formats or self.app.builder.format in self.formats
```

File: sphinx/ext/autodoc/__init__.py [hunk 3]
```diff
@@ -461,9 +461,7 @@ def check_module(self) -> bool:
 
         subject = inspect.unpartial(self.object)
         modname = self.get_attr(subject, '__module__', None)
-        if modname and modname != self.modname:
-            return False
-        return True
+        return not (modname and modname != self.modname)
```
Comment (picnixz) [inline sphinx/ext/autodoc/__init__.py]:
```suggestion
        return not modname or modname == self.modname
```

**Review by picnixz — APPROVED** [review]

**Comment by picnixz** [comment]
I'm still a bit sleepy so could you verify (again) that my boolean conditions are correct (I don't want to miss anything)

**Comment by danieleades** [comment]
> I'm still a bit sleepy so could you verify (again) that my boolean conditions are correct (I don't want to miss anything)

they look fine to me. De Morgan's laws for the win!

**Comment by picnixz** [comment]
Thank you!

**Pull request merged**